### PR TITLE
S3C-184 TEST: add getObject versioning tests

### PR DESCRIPTION
**Pull request by nicolas2bert**
Add more tests for getObject versioning.

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    assert.notEqual(err, null,
+        'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+    assert.strictEqual(err.statusCode, statusCode);
+}
+
+const testing = process.env.VERSIONING === 'no' ? describe.skip : describe;
+
+testing('get object with versioning', function testSuite() {
+    this.timeout(600000);
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: please remove

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                    Key: key,
+                    VersionId: this.test.versionId },
+                (err, data) => {
+                    _assertNoError(err, 'getObject');
+                    assert.strictEqual(data.VersionId, this.test.versionId);
+                    done();
+                });
+            });
+
+            it('should not be able to get an object that has been delete ' +
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: it should return NoSuchKey if get object without version and latest version is a delete marker

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                    Bucket: bucket,
+                    Key: key,
+                    VersionId: this.test.deleteVersionId },
+                err => {
+                    _assertError(err, 405, 'MethodNotAllowed');
+                    done();
+                });
+            });
+
+            it('should not be able to get a delete marker without ' +
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: it should return NoSuchKey if try to get object whose latest version is a delete marker

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                    Bucket: bucket,
+                    Key: key,
+                    VersionId: this.test.deleteVersionId },
+                err => {
+                    _assertError(err, 405, 'MethodNotAllowed');
+                    done();
+                });
+            });
+
+            it('should be able to get an object using its versionId',
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: it should be able to get a version that was put prior to the delete marker

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                    Bucket: bucket,
+                    Key: key },
+                err => {
+                    _assertError(err, 404, 'NoSuchKey');
+                    done();
+                });
+            });
+        });
+
+        describe('getting a object that has been deleted', () => {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: behavior when put version with content then put delete marker

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+function _assertError(err, statusCode, code) {
+    assert.notEqual(err, null,
+        'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+    assert.strictEqual(err.statusCode, statusCode);
+}
+
+const testing = process.env.VERSIONING === 'no' ? describe.skip : describe;
+
+testing('get object with versioning', function testSuite() {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: get behavior after delete with versioning

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        });
+
+        afterEach(done => {
+            removeAllVersions({ Bucket: bucket }, err => {
+                _assertNoError(err, 'removeAllVersions');
+                return s3.deleteBucket({ Bucket: bucket }, done);
+            });
+        });
+
+        describe('get a delete marker', () => {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: behavior when only version put is a delete marker

**Comment by nicolas2bert** [comment]
@ironman-machine try

**Pull request merged**